synchronizer/l2_sync/l2_sync_etrog/executor_trusted_batch_sync.go

```diff
@@ -222,17 +349,64 @@ func checkThatL2DataIsIncremental(data *l2_shared.ProcessData) error {
 	incommingData := data.TrustedBatch.BatchL2Data
 	previousData := data.StateBatch.BatchL2Data
 	if len(incommingData) < len(previousData) {
-		return fmt.Errorf("the new batch has less data than the one in node err:%w", ErrBatchDataIsNotIncremental)
+		return fmt.Errorf("L2Data check: the new batch has less data than the one in node err:%w", ErrBatchDataIsNotIncremental)
 	}
-	if len(incommingData) == len(previousData) {
-		return fmt.Errorf("the new batch has the same data than the one in node err:%w", ErrBatchDataIsNotIncremental)
-	}
-	if hash(incommingData) != hash(previousData) {
-		return fmt.Errorf("the new batch has different data than the one in node err:%w", ErrBatchDataIsNotIncremental)
+
+	if hash(incommingData[:len(previousData)]) != hash(previousData) {
+		findFirstByteDifferent := findFirstByteDifferent(incommingData, previousData)
+		log.Info(syncCommon.LogComparedBytes("trusted L2BatchData", "state   L2BatchData", incommingData, previousData, 10, 10)) //nolint:gomnd
+		return fmt.Errorf("L2Data check: the common part with state dont have same hash (first byte different at: %d) err:%w", findFirstByteDifferent, ErrBatchDataIsNotIncremental)
 	}
 	return nil
 }
 
+func findFirstByteDifferent(data1 []byte, data2 []byte) int {
+	for i := 0; i < len(data1); i++ {
+		if data1[i] != data2[i] {
+			return i
+		}
+	}
+	return -1
+}
+
+func sumAllL2BlockDeltaTimestamp(rawBatch *state.BatchRawV2) uint32 {
+	var sum uint32 = 0
+	for _, l2block := range rawBatch.Blocks {
+		sum += l2block.DeltaTimestamp
+	}
+	return sum
+}
+
+func (b *SyncTrustedBatchExecutorForEtrog) composePartialBatch(previousBatch *state.Batch, newBatch *types.Batch) (types.ArgBytes, error) {
+	debugStr := " composePartialBatch: "
```

Comment:
You should use the context instead of this workaround. or At least use logger.WithValues

Reply:
I have created a issue to do that: https://github.com/0xPolygonHermez/zkevm-node/issues/2952